src/test/java/com/microsoft/sqlserver/jdbc/AlwaysEncrypted/AESetup.java

```diff
@@ -1959,4 +1941,38 @@ private static void dropCMK(String cmkName, Statement stmt) throws SQLException
                 + " drop column master key " + cmkName + " end";
         stmt.execute(cekSql);
     }
+
+    /**
+     * Alter Column encryption on deterministic columns to randomized - this will trigger enclave to re-encrypt
+     * 
+     * @param stmt
+     * @param tableName
+     * @param table
+     * @param values
+     * @throws SQLException
+     */
+    protected void testAlterColumnEncryption(SQLServerStatement stmt, String tableName, String table[][],
+            String cekName) throws SQLException {
+        try (SQLServerConnection con = PrepUtil.getConnection(AETestConnectionString, AEInfo)) {
+            for (int i = 0; i < table.length; i++) {
+                // alter deterministic to randomized
+                String sql = "ALTER TABLE " + tableName + " ALTER COLUMN " + ColumnType.DETERMINISTIC.name()
+                        + table[i][0] + " " + table[i][1]
+                        + String.format(encryptSql, ColumnType.RANDOMIZED.name(), cekName) + ")";
+                try (SQLServerPreparedStatement pstmt = (SQLServerPreparedStatement) TestUtils.getPreparedStmt(con, sql,
+                        stmtColEncSetting)) {
+                    stmt.execute(sql);
+                    if (!TestUtils.isAEv2(con)) {
+                        fail(TestResource.getResource("R_expectedExceptionNotThrown"));
+                    }
+                } catch (SQLException e) {
+                    if (!TestUtils.isAEv2(con)) {
+                        fail(TestResource.getResource("R_expectedExceptionNotThrown"));
```

Comment:
This code throws an exception with error message `"Expected exception is not thrown."`, which might be misleading because SQLServerException will get thrown when the server does not support AEv2, which is expected. Perhaps it is better to fail with `fail(e.getMessage());`.

Reply:
good point, please make the same change in JDBCEncryptionDecryptionTest